app/src/main/java/com/bignerdranch/android/criminalintent/CrimeAdapter.kt

```diff
@@ -25,16 +31,19 @@ class CrimeAdapter(var crimes: List<Crime>) : RecyclerView.Adapter<CrimeAdapter.
 
     private val titleTextView: TextView = itemView.findViewById(R.id.crimeTitle)
     private val dateTextView: TextView = itemView.findViewById(R.id.crimeDate)
+    private val solvedImageView: ImageView = itemView.findViewById(R.id.crimeSolved)
 
     fun bind(crime: Crime) {
       titleTextView.text = crime.title
-      dateTextView.text = crime.date.toString()
+      dateTextView.text = DateFormat.format(DATE_FORMAT, crime.date)
+
       with(itemView) {
         var text = context.getString(R.string.crime_title_pressed, crime.title)
         setOnClickListener {
           Toast.makeText(context, text, Toast.LENGTH_SHORT).show()
         }
       }
+      solvedImageView.visibility = if (crime.isSolved) View.VISIBLE else View.GONE
```

Comment:
include android kotlin core extensions and use `solvedImageView.isVisible = crime.isSolved`